### PR TITLE
Wean off IOLoop.run_sync

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Add some more changes in preparation for tornado 5.1.1

### What issues does this PR fix or reference?
Fixes:

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by dwoz** [comment]
re-run full all

File: salt/utils/event.py [hunk 1]
```diff
@@ -674,7 +676,7 @@ def get_event_noblock(self):
         if not self.cpub:
             if not self.connect_pub():
                 return None
-        raw = self.subscriber.read_sync(timeout=0)
+        raw = self.subscriber._read(timeout=0)
```
Comment (s0undt3ch) [inline salt/utils/event.py:679]: Are you sure it's `_read` here?

Comment (dwoz) [reply salt/utils/event.py:679]: Yes, this is intentional.

Comment (s0undt3ch) [reply salt/utils/event.py:679]: Ok.

**Review by s0undt3ch — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Pull request merged**